### PR TITLE
Add egress diagram to docs

### PR DESCRIPTION
**Pull request by SimonBaeumer**
## Description
Add egress diagram to docs

## Checklist (Definition of Done)

- [ ] Unit and integration tests added
- [ ] Added test description under `Test manual`
- [ ] Documentation added if necessary (i.e. changes to dev setup, test execution, ...)
- [ ] CI and all relevant tests are passing
- [ ] Add the ticket number to the PR title if available, i.e. `ROX-12345: ...`
- [ ] Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.
- [ ] Add secret to app-interface Vault or Secrets Manager if necessary
- [ ] RDS changes were e2e tested [manually](../docs/development/howto-e2e-test-rds.md)
- [ ] Check AWS limits are reasonable for changes provisioning new resources

## Test manual

**TODO:** Add manual testing efforts

```
# To run tests locally run:
make db/teardown db/setup db/migrate
make ocm/setup OCM_OFFLINE_TOKEN=<ocm-offline-token> OCM_ENV=development
make verify lint binary test test/integration
```

### CODE REVIEW

File: docs/architecture/images/egress_diagram.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    direction LR
+    subgraph RDS
+            acs1rdsp["RDS 1 R/W Primary 🐘⛁"]
+    end
+
+    subgraph Data Plane OSD Cluster
+
+        subgraph acs-namespace
+            acs1central[Central]
+            acs1egress[egress]
```
Comment (johannes94) [inline docs/architecture/images/egress_diagram.md:13]: If I understand correctly this is the egress-proxy? I would change the names in this diagram so it matches the names of k8s resources e.g egress -> egress-proxy

**Review by johannes94 — CHANGES_REQUESTED** [review]
Except the one comment the diagram looks good.

**Review by ebensh — DISMISSED** [review]

File: docs/architecture/images/egress_diagram.md (hunk 1 above)
Comment (SimonBaeumer) [reply docs/architecture/images/egress_diagram.md]: Updated

File: docs/architecture/images/egress_diagram.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            acs1scanner[Scanner]
+        end
+
+        acs1central-->acs1scanner
+        acs1central-->acs1egress
```
[the remark this answers is not in the document]
Comment (SimonBaeumer) [reply docs/architecture/images/egress_diagram.md:18]: Good catch, Scanner uses the egress-proxy too. Fixed the diagram

**Comment by johannes94** [comment]
@SimonBaeumer was adding the values.yaml file intentional?

**Comment by SimonBaeumer** [comment]
> @SimonBaeumer was adding the values.yaml file intentional?

@johannes94 Fixed

File: docs/architecture/images/egress_diagram.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (johannes94) [reply docs/architecture/images/egress_diagram.md:18]: In the current diagram scanner is pointing to the internet without going through the egress-proxy first.

**Review by johannes94 — CHANGES_REQUESTED** [review]

File: docs/architecture/images/egress_diagram.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (SimonBaeumer) [reply docs/architecture/images/egress_diagram.md:18]: Fixed!

**Review by johannes94 — APPROVED** [review]

**Pull request merged**